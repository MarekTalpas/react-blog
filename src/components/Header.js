import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

export const Header = ({ startLogout }) => (
  <header className="header">
    <div className="content-container">
      <div className="header__content">
        <Link className="header__title" to="/dashboard">
          <h1>React Blog</h1>
        </Link>
        <button className="button button--link" onClick={startLogout}>Logout</button>
      </div>
    </div>
  </header>
);

const mapDispatchToProps = dispatch => ({
  startLogout: () => dispatch(startLogout())
});

Header.propTypes = {
  startLogout: PropTypes.func.isRequired
};

export default connect(undefined, mapDispatchToProps)(Header);
