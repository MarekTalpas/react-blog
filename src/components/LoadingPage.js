import React from 'react';

const LoadingPage = () => (
  <div className="loader">
    <img className="loader__image" src="/images/loader.gif" alt="circle loader" />
  </div>
);

export default LoadingPage;
