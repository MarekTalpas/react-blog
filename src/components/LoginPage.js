import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { startLoginWithGoogle, startLoginWithFacebook } from '../actions/auth';

export const LoginPage = ({ startLoginWithGoogle, startLoginWithFacebook }) => (
  <div className="box-layout">
    <div className="box-layout__box">
      <h1 className="box-layout__title">React Blog</h1>
      <p className="box-layout__subtitle">Best Blog ever.</p>
      <button className="button google" onClick={startLoginWithGoogle}>Login with Google</button>
      <button className="button facebook" onClick={startLoginWithFacebook}>Login with Facebook</button>
    </div>
  </div>
);

LoginPage.defaultProps = {
  startLoginWithGoogle: () => {},
  startLoginWithFacebook: () => {}
};

LoginPage.propTypes = {
  startLoginWithGoogle: PropTypes.func,
  startLoginWithFacebook: PropTypes.func
};

const mapDispatchToProps = dispatch => ({
  startLoginWithGoogle: () => dispatch(startLoginWithGoogle()),
  startLoginWithFacebook: () => dispatch(startLoginWithFacebook())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
