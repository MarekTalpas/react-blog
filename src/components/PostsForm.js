import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { SingleDatePicker } from 'react-dates';
import moment from 'moment';

class PostsForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
      title: props.post ? props.post.title : '',
      createdAt: props.post ? moment(props.post.createdAt) : moment(),
      content: props.post ? props.post.content : '',
      calendarFocused: false,
      error: ''
    };
  };
  _onTitleChange = ev => {
    const title = ev.target.value;
    this.setState(() => ({ title }));
  };
  _onMainContentChange = ev => {
    const content = ev.target.value;
    this.setState(() => ({ content }));
  };
  _onCalendarFocusChange = ({ focused }) => {
    this.setState(() => ({ calendarFocused: focused }));
  };
  _onDateChange = date => this.setState(() => ({ createdAt: date }));
  _onFormSubmit = (ev) => {
    ev.preventDefault();

    if (!this.state.title || !this.state.content) {
      this.setState(() => ({ error: 'You have to provide content and title' }));
    } else {
      this.setState(() => ({ error: '' }));
      const post = {
        title: this.state.title,
        createdAt: this.state.createdAt.valueOf(),
        content: this.state.content
      }
      this.props.onSubmit(post);
    }
  };
  render() {
    return (
      <form className="form" onSubmit={this._onFormSubmit}>
        { this.state.error && <p>{this.state.error}</p> }
        <input
          type="text"
          className="text-input"
          placeholder="Title"
          autoFocus
          value={this.state.title}
          onChange={this._onTitleChange}
        />
        <SingleDatePicker
          date={this.state.createdAt}
          onDateChange={this._onDateChange}
          focused={this.state.calendarFocused}
          onFocusChange={this._onCalendarFocusChange}
          numberOfMonths={1}
          isOutsideRange={() => false}
        />
        <textarea
          className="textarea"
          placeholder="Provide your post content here"
          value={this.state.content}
          onChange={this._onMainContentChange}
        />
        <div>
          <button className="button">Save Post</button>
        </div>
      </form>
    );
  }
}

PostsForm.defaultProps = {
  post: {
    title: '',
    createdAt: moment().valueOf(),
    content: ''
  },
  onSubmit: () => { }
}

PostsForm.propTypes = {
  post: PropTypes.shape({
    title: PropTypes.string,
    createdAt: PropTypes.number,
    content: PropTypes.string
  }),
  onSubmit: PropTypes.func
};

export default PostsForm;
