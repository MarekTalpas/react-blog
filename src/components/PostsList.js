import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import PostsListItem from './PostsListItem';
import filterPosts from '../selectors/posts';

export const PostsList = ({ posts }) => (
  <div className="content-container">
    <div className="list-header">
      <div className="show-for-mobile">Posts</div>
      <div className="show-for-desktop">Posts</div>
    </div>
    <div className="list-body">
      {
        posts.length === 0 ? (
          <div className="list-item list-item--message">
            <span>No Posts To Show</span>
          </div>
        ) : (
          posts.map(post => <PostsListItem key={post.id} {...post} />)
        )
      }
    </div>
  </div>
);

PostsList.propTypes = {
  posts: PropTypes.shape({
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    content: PropTypes.string.isRequired,
    createdAt: PropTypes.number.isRequired
  }).isRequired
};

const mapStateToProps = state => ({
  posts: filterPosts(state.posts, state.filters)
});

export default connect(mapStateToProps)(PostsList);
