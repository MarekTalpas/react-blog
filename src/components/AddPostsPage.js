import React, { Component } from 'react';
import { connect } from 'react-redux';
import PostsForm from './PostsForm';
import { initiateAddPost } from '../actions/posts';

export class AddPostsPage extends Component {
  _handleFormSubmit = (post) => {
    this.props.initiateAddPost(post);
    this.props.history.push('/');
  };
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Create Post</h1>
          </div>
        </div>
        <div className="content-container">
          <PostsForm onSubmit={this._handleFormSubmit} />
        </div>
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  initiateAddPost: (post) => dispatch(initiateAddPost(post))
});

export default connect(undefined, mapDispatchToProps)(AddPostsPage);
