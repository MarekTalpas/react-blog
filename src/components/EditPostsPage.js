import React, { Component } from 'react';
import Modal from 'react-modal';
import { connect } from 'react-redux';
import PostsForm from './PostsForm';
import { initiateEditPost, initiateRemovePost } from '../actions/posts';

window.Modal = Modal;

const modalStyles = {
  overlay: {
    display: 'flex',
    alignItems: 'center',
    height: '100vh',
    justifyContent: 'center',
    width: '100vw'
  },
  content: {
    borderRadius: '3px',
    boxShadow: '0 14px 28px rgba(0, 0, 0, 0.25), 0 10px 10px rgba(0, 0, 0, 0.22)',
    margin: '$l-size $m-size',
    position: 'none',
    textAlign: 'center',
    width: '40rem'
  }
};

export class EditPostsPage extends Component {
  state = {
    modalIsOpen: false
  };
  _openModal = () => {
    this.setState(() => ({ modalIsOpen: true }));
  };
  _closeModal = () => {
    this.setState(() => ({ modalIsOpen: false }));
  };
  _handleFormSubmit = (post) => {
    this.props.initiateEditPost(this.props.post.id, post);
    this.props.history.push('/');
  };
  _handlePostRemove = () => {
    this.props.initiateRemovePost({ id: this.props.post.id });
    this.props.history.push('/');
  };
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Edit Post</h1>
          </div>
        </div>
        <div className="content-container">
          <PostsForm onSubmit={this._handleFormSubmit} post={this.props.post} />
          <button
            className="button button--secondary button--remove"
            onClick={this._openModal}
          >
            Remove Post
          </button>
        </div>
        <Modal
          isOpen={this.state.modalIsOpen}
          onRequestClose={this._closeModal}
          closeTimeoutMS={400}
          style={modalStyles}
          contentLabel="Remove Post"
        >
          <h1 className="box-layout__title">Do you really want to remove this post ?</h1>
          <div className="button__couple">
            <button
              className="button button--secondary button--confirm"
              onClick={this._closeModal}
            >
              NO
            </button>
            <button
              className="button button--confirm"
              onClick={this._handlePostRemove}
            >
              YES
            </button>
          </div>
        </Modal>
      </div>
    );
  }
}

const mapStateToProps = (state, props) => ({
  post: state.posts.find(post => post.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch) => ({
  initiateEditPost: (post) => dispatch(initiateEditPost(post)),
  initiateRemovePost: (id) => dispatch(initiateRemovePost(id))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditPostsPage);
