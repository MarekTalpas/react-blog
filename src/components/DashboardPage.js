import React from 'react';
import PostsListFilters from './PostsListFilters';
import PostsList from './PostsList';

const DashboardPage = () => (
  <div className="header--dashboard">
    <PostsListFilters />
    <PostsList />
  </div>
);

export default DashboardPage;
