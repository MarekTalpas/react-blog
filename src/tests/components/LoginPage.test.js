import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../components/LoginPage';

test('should correctly render LoginPage', () => {
  const wrapper = shallow(<LoginPage />);
  expect(wrapper).toMatchSnapshot();
});

test('should call startLoginWithGoogle on google button click', () => {
  const startLoginWithGoogle = jest.fn();
  const wrapper = shallow(<LoginPage
    startLoginWithGoogle={startLoginWithGoogle}
  />);
  wrapper.find('.google').simulate('click');
  expect(startLoginWithGoogle).toHaveBeenCalled();
});

test('should call startLoginWithFAcebook on facebook button click', () => {
  const startLoginWithFacebook = jest.fn();
  const wrapper = shallow(<LoginPage
    startLoginWithFacebook={startLoginWithFacebook}
  />);
  wrapper.find('.facebook').simulate('click');
  expect(startLoginWithFacebook).toHaveBeenCalled();
});
