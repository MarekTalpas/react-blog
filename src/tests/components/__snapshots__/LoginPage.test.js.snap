// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render LoginPage 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      React Blog
    </h1>
    <p
      className="box-layout__subtitle"
    >
      Best Blog ever.
    </p>
    <button
      className="button google"
      onClick={[Function]}
    >
      Login with Google
    </button>
    <button
      className="button facebook"
      onClick={[Function]}
    >
      Login with Facebook
    </button>
  </div>
</div>
`;
